commcare/src/main/java/org/motechproject/commcare/web/ConfigController.java

```diff
@@ -72,6 +72,17 @@ public void verifyConfig(@RequestBody Config config) {
     }
 
     @ResponseStatus(HttpStatus.OK)
+    @RequestMapping(value="/sync", method = RequestMethod.POST)
+    @ResponseBody
+    public void syncConfig(@RequestBody Config config) {
+        if (configService.verifyConfig(config)) {
+            configService.syncConfig(config.getName());
+        } else {
+            throw new CommcareAuthenticationException("Motech was unable to authenticate to CommCareHQ. Please verify your account settings.");
```

Comment:
Can we refactor this exception to include the message inside, rather than having it passed wherever it is used? We seem to pass various messages that basically state the same thing.